docs/sources/flow/reference/components/pyroscope.ebpf.md

```diff
@@ -0,0 +1,246 @@
+* `elf_cache` per build id and per same file symbol tables and their sizes in symbols count.
+
+## Debug metrics
+
+* `pyroscope_fanout_latency` (histogram): Write latency for sending to direct and indirect components.
+* `pyroscope_ebpf_active_targets` (gauge): Number of active targets the component tracks.
+* `pyroscope_ebpf_profiling_sessions_total` (counter): Number of profiling sessions completed.
+* `pyroscope_ebpf_profiling_sessions_failing_total` (counter): Number of profiling sessions failed.
+* `pyroscope_ebpf_pprofs_total` (counter): Number of pprof profiles collected by the ebpf component.
+
+## Profile collecting behavior
+
+The `pyroscope.ebpf` component collects stack traces associated with a process running on the current host.
+You can use the `sample_rate` argument to define the number of stack traces collected per second. The default is 97.
+
+The following labels are automatically injected into the collected profiles if you have not defined them. These labels
+can help you pin down a profiling target.
+
+| Label              | Description                                                                                                                      |
+|--------------------|----------------------------------------------------------------------------------------------------------------------------------|
+| `service_name`     | Pyroscope service name. It's automatically selected from discovery meta labels if possible. Otherwise defaults to `unspecified`. |
+| `__name__`         | pyroscope metric name. Defaults to `process_cpu`.                                                                                |
+| `__container_id__` | The container ID derived from target.                                                                                            |
+
+### Container ID
+
+Each collected stack trace is then associated with a specified target from the targets list, determined by a
+container ID. This association process involves checking the `__container_id__`, `__meta_docker_container_id`,
+and `__meta_kubernetes_pod_container_id` labels of a target against the `/proc/{pid}/cgroup` of a process.
+
+If a corresponding container ID is found, the stack traces are aggregated per target based on the container ID.
+If a container ID is not found, the stack trace is associated with a `default_target`.
+
+Any stack traces not associated with a listed target are ignored.
+
+### Service name
+
+The special label `service_name` is required and must always be present. If it's not specified, it is
+attempted to be inferred from multiple sources:
+
+- `__meta_kubernetes_pod_annotation_pyroscope_io_service_name` which is a `pyroscope.io/service_name` pod annotation.
+- `__meta_kubernetes_namespace` and `__meta_kubernetes_pod_container_name`
+- `__meta_docker_container_name`
+
+If `service_name` is not specified and could not be inferred, it is set to `unspecified`.
+
+## Troubleshooting Unknown Symbols
```

Comment:
Docs recommendation are to use sentence case for headers:

```suggestion
## Troubleshooting unknown symbols
```

Reply:
Ooops I missed that in my review. :-P